### ISSUE TITLE
Allow specifying the label angle, font, justification, and offset in pygmt.meca

### ISSUE THREAD

**Issue by meghanrjones** [GenericMappingTools/pygmt#1466]
**Description of the desired feature**

Based on [this forum post](https://forum.generic-mapping-tools.org/t/pygmt-how-to-change-meca-label-sizes/2023), it would be useful to add support for customizing the label characteristics for meca, which is done using **-S** in [gmt meca](https://docs.generic-mapping-tools.org/dev/supplements/seis/meca.html#s). Relates to https://github.com/GenericMappingTools/pygmt/issues/1082, in order to make the PyGMT implementation more pythonic than gmt meca **-S**. 

**Are you willing to help implement and maintain this feature?** Not likely that I will find time to implement

**Comment by weiji14** [comment]
Note that there was a previous attempt to handle the text label specification. See @tjnewton's implementation at 2fe87af5adb9403ee17d388d63fb14cf351be1fa, but this was reverted by @liamtoney at ff164e654827024e76ec892f3acc2ee0d4c47bef because we didn't support `GMT_Put_Strings` at that time (full explanation at https://github.com/GenericMappingTools/pygmt/pull/516#issuecomment-657313059.

I'll try and resurrect that old bit of code, and put it in #1613. Or, I could implement the text label and angle/font/justification specification in a separate Pull Request after #1613.
